daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/ScenarioRunner.scala

```diff
@@ -258,22 +258,22 @@ object ScenarioRunner {
     ): Either[SError, R]
   }
 
-  case class ScenarioLedgerApi(ledger: ScenarioLedger)
-      extends LedgerApi[ScenarioLedger.CommitResult] {
+  class ScenarioLedgerApi(ledger: ScenarioLedger) extends LedgerApi[ScenarioLedger.CommitResult] {
```

Comment:
why? for something immutable like this one a case class seems reasonable to me.

Reply:
I do not like it contains state full data. 
I will revert it anyway.

Reply:
`ScenarioLedger` is immutable. But perfectly fine to keep it, just wanted to understand.

Reply:
my mistake.